processor/src/main/java/org/mapstruct/ap/internal/processor/MapperCreationProcessor.java

```diff
@@ -294,10 +296,20 @@ else if ( constructor.getParameters().size() == 1 ) {
         return decorator;
     }
 
-    private SortedSet<Type> getExtraImports(TypeElement element,  MapperOptions mapperOptions) {
+    private SortedSet<Type> getExtraImports(TypeElement element, MapperOptions mapperOptions,
+                                            List<SourceMethod> methods) {
         SortedSet<Type> extraImports = new TreeSet<>();
 
 
+        for ( SourceMethod method : methods ) {
+            for ( SubClassMappingOptions subClassMapping : method.getOptions().getSubClassMappings() ) {
+                Type type = typeFactory.getType( subClassMapping.getTargetClass() );
+                extraImports.add( type );
+                type = typeFactory.getType( subClassMapping.getSourceClass() );
+                extraImports.add( type );
+            }
+        }
```

Comment:
I do not think that this is needed. The `SubClassMapping` already implements the `getImportTypes`, so you only need to call that for each `subClassMapping` in the `BeanMappingMethod#getImportTypes`